### PR TITLE
8266518: Refactor and expand scatter/gather tests

### PR DESCRIPTION
**Pull request by PaulSandoz**
Refactor scatter/gather tests to be included in the load/store test classes and expand to support access between `ShortVector` and and `char[]`, and access between `ByteVector` and `boolean[]`.

Vector tests pass on linux-x64 linux-aarch64 macosx-x64, and windows-x64.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266518](https://bugs.openjdk.java.net/browse/JDK-8266518): Refactor and expand scatter/gather tests

### Reviewers
 * [Sandhya Viswanathan](https://openjdk.java.net/census#sviswanathan) (@sviswa7 - **Reviewer**) ⚠️ Review applies to 6455ae437ea5aa1e681fcffe818f5723fa815667

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/48/head:pull/48` \
`$ git checkout pull/48`

Update a local copy of the PR: \
`$ git checkout pull/48` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/48/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 48`

View PR using the GUI difftool: \
`$ git pr show -t 48`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/48.diff">https://git.openjdk.java.net/jdk17/pull/48.diff</a>

</details>

### CODE REVIEW

File: test/jdk/jdk/incubator/vector/Byte128VectorLoadStoreTests.java [hunk 1]
```diff
@@ -1038,4 +1018,301 @@ static void loadStoreMaskBooleanArray(IntFunction<byte[]> fa,
...
+    @DataProvider
+    public Object[][] byteGatherScatterMaskProvider() {
+        return BOOLEAN_MASK_GENERATORS.stream().
+          flatMap(fs -> INT_INDEX_GENERATORS.stream().flatMap(fm ->
+            BYTE_GENERATORS.stream().map(fa -> {
+                    return new Object[] {fa, fm, fs};
+            }))).
+            toArray(Object[][]::new);
+    }
+
```
Comment (sviswa7) [inline test/jdk/jdk/incubator/vector/Byte128VectorLoadStoreTests.java:1248]: The byteGatherScatterProvider and byteGatherScatterMaskProvider seem to be same as gatherScatterProvider and gatherScatterMaskProvider above.

**Review by sviswa7 — APPROVED** [review]
Thanks for refactoring and adding new scatter/gather tests for boolean and char. Other than the duplicate data provider in byte/boolean tests, the rest looks good to me.

File: test/jdk/jdk/incubator/vector/Byte128VectorLoadStoreTests.java (hunk 1 above)
Comment (PaulSandoz) [reply test/jdk/jdk/incubator/vector/Byte128VectorLoadStoreTests.java]: Well spotted. I removed the data providers.

**Comment by sviswa7** [comment]
Looks good to me.

**Pull request closed without merging**